File: exercise/day12/src/main/java/greeting/Greeter.java
package greeting;

public class Greeter {
    String formality;

    public String greet() {
        if (this.formality == null) {
            return "Hello.";
        }

        if (this.formality.equals("formal")) {
            return "Good evening, sir.";
        } else if (this.formality.equals("casual")) {
            return "Sup bro?";
        } else if (this.formality.equals("intimate")) {
            return "Hello Darling!";
        } else {
            return "Hello.";
        }
    }

    public void setFormality(String formality) {
        this.formality = formality;
    }
}

File: exercise/kotlin/day12/src/main/kotlin/greeting/Greeter.kt
package greeting

class Greeter(private val formality: String? = null) {
    fun greet(): String {
        if (formality == null) {
            return "Hello."
        }
        return if (formality == "formal") {
            "Good evening, sir."
        } else if (formality == "casual") {
            "Sup bro?"
        } else if (formality == "intimate") {
            "Hello Darling!"
        } else {
            "Hello."
        }
    }
}